src/components/Layout/Header/menu/HeaderSubMenu.tsx

```diff
@@ -30,6 +35,14 @@ export const HeaderSubMenu = ({ item }: Props) => {
   const { key, subItems } = item
   const isActive = subItems.some(({ href }) => match({ to: href }))
 
+  const {
+    current: { pathname },
+  } = useLocation()
+
+  useEffect(() => {
```

Comment:
Wouldn't it better to add onClick handler to the Link when you click it, just close the popup?

Reply:
I agree. This also won't solve a case when user clicks on a menu item that links to the current page.

Reply:
updated